### PR TITLE
Update react-native-bridge to build an AAR to publish on Bintray and use in WordPress Android

### PR DESCRIPTION
**Pull request by mokagio**
## Description

This is part of the [Composite Android build for Gutenberg](https://github.com/orgs/wordpress-mobile/projects/80#card-52385844), an effort to simplify and speed up the build and workflow for developers of the WordPress Android app.

The changes in this PR are the minimum require to build an AAR that can both be published to Bintray and used by the WordPress app.

The diff includes changes to the `package-lock.json` because [CI told me so](https://github.com/WordPress/gutenberg/pull/28216/checks?check_run_id=1710376166) even though I didn't changed the `package.json` itself 🤔 .

This supersedes #27836.

## How has this been tested?

I generated and published an AAR from this changeset and verified it works in the WordPress app via [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13642). 

## Types of changes

This PR tweaks the `react-native-bridge/android` build configuration only.

## Checklist:

- [ ] ~~My code is tested.~~ There's no infrastructure to test these changes automatically. I only performed manual testing, i.e.: run the generated AAR in the WrodPress Android app and verified it works.
- [ ] ~~My code follows the WordPress code style.~~ N.A. 
- [ ] ~~My code follows the accessibility standards.~~ N.A. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] ~~I've updated all React Native files affected by any refactorings/renamings in this PR.~~  N.A.

### CODE REVIEW

File: packages/react-native-bridge/android/build.gradle [hunk 1]
```diff
@@ -341,3 +379,64 @@ If they are changed, the isBundleUpToDate flag is switched to false. That flag i
...
+    publications = ['GutenbergMobilePublication']
+    publish = true
+    pkg {
+        repo = 'maven'
+        name = 'react-native-gutenberg-bridge'
+        licenses = ['GPL-2.0']
+        // FIXME: Once this works properly poit to 'wordpress-mobile'
+        userOrg = 'mokagio'
+        // FIXME: Once this works properly poit to the original gutenberg (or gutenberg-mobile) repo. See also: https://github.com/oguzkocer/gutenberg/commit/0ca71b1abe3ea22b77abb3f390534e66e39161d5#diff-320e7dacc9e9a83ac8128d299319f82cf6dfa2609e0ebbc043d13de33ea92773R350-R353
+        vcsUrl = 'https://github.com/wordpress-mobile/gutenberg-mobile.git'
```
Comment (mokagio) [inline packages/react-native-bridge/android/build.gradle:401]: I discussed this with @oguzkocer and @mchowning. I'll update to point to `wordpress-mobile` before merging this.

File: packages/react-native-bridge/android/build.gradle [hunk 2]
```diff
@@ -114,6 +120,11 @@ android {
     sourceSets {
         main {
             assets.srcDirs += buildAssetsFolder
+            assets.srcDirs += '../../../../src/block-support'
+            // Despite being in a folder called "resources", the files in
+            // unsupported-block-editor are accessed as assets by their
+            // consumers: the WordPressEditor library.
+            assets.srcDirs += '../../../../resources/unsupported-block-editor'
```
Comment (mokagio) [inline packages/react-native-bridge/android/build.gradle:127]: These are not _required_ for the AAR generation, but we'll use them to address https://github.com/wordpress-mobile/WordPress-Android/issues/13747.

**Comment by mokagio** [comment]
I'm concerned by the Android E2E tests failing. @mchowning any tips on how to start looking into it?

**Comment by mchowning** [comment]
This is looking good. I tested it and it is generally working quite well! 👍  The only issue I'm seeing is that, unless I'm mistaken, the build is not generating the javascript bundle by default. So if you `clean`, the aar that is generated is like ~200kb (instead of >3mb) because there is no js bundle. Once you generate the js bundle once, you won't have this problem unless you run the clean task because it is cached (or if you were to change the js code, you wouldn't see your changes when you used the resulting aar). You should be able to reproduce this by running `./gradlew clean assembleRelease` and then looking at the resulting aar. To get the bundle generation we need to set the `buildGutenbergFromSourceFlag` by doing something like this, which works for me:

```
./gradlew clean assembleRelease -PbuildGutenbergMobileJSBundle=true
```

I realize that this setup is going to change a lot, particularly with @oguzkocer 's composite build changes, so I don't think we need to have this fully worked out at this point. With that said, I think we would be better off building the js bundle default, and requiring that we affirmatively do something (set a variable, etc.) to tell the build to skip building the js bundle. wdyt?

> I'm concerned by the Android E2E tests failing.

Our E2E tests can be flaky 😞  (although this seems to be failing somewhat consistently). The tests passed for me locally, so I think it may not be worth worrying about them at this point since we're only merging to a feature branch. I also don't think the performance test failures have anything to do with your changes. I think there's a decent chance that once we bring in the latest changes from `master` the CI issues will resolve themselves.

I don't view any of these issues as blockers since we're targeting a feature branch, so I'll go ahead and approve. Just let me (or, if I'm not around, any other gutenberg dev) know if you want to get this merged (I'm assuming you won't be able to merge it yourself yet).

**Review by mchowning — APPROVED** [review]

File: packages/react-native-bridge/android/build.gradle (hunk 2 above)
Comment (mchowning) [reply packages/react-native-bridge/android/build.gradle:127]: Right, I assume this is something we'll take another look at before merging to `master`.

**Comment by mchowning** [comment]
Going ahead and merging this so @oguzkocer can work on top of this.

**Pull request merged**